drivers/gpio/gpio_pca_series.c

```diff
@@ -0,0 +1,2077 @@
+					"reg cache",
+					reg_type);
+				reg = PCA_REG_INVALID;
+			}
+		}
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+
+		/** do_print */
+		LOG_INF("%.2d\t"
+			"%s\t"
+			"0x%2.2x\t"
+#ifdef CONFIG_GPIO_PCA_SERIES_CACHE_ALL
+			"0x%2.2x\t"
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+			,
+			reg_type, gpio_pca_series_reg_name[reg_type], reg
+#ifdef CONFIG_GPIO_PCA_SERIES_CACHE_ALL
+			,
+			cache
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+		);
+	}
+}
+
+/**
+ * @brief configure output drive strength
+ *
+ * @param dev
+ * @param pin
+ * @param strength
+ * @return int 0 if success
+ *             -EIO if i2c error
+ */
+static int gpio_pca_series_pin_config_drive_strength(
+		const struct device *dev, gpio_pin_t pin,
+		enum gpio_pca_series_drive_strength strength)
+{
+	const struct gpio_pca_series_config *cfg = dev->config;
+	struct gpio_pca_series_data *data = dev->data;
+	int ret;
+	uint32_t reg_cfg_shift = pin << 1U;
+	uint64_t reg_cfg_mask = 0b11 << reg_cfg_shift;
+	uint64_t reg_value;
+
+	if ((cfg->part_cfg->flags & PCA_HAS_LATCH) == 0U) {
+		return -ENOTSUP;
+	}
+	/* Can't do I2C bus operations from an ISR */
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	k_sem_take(&data->lock, K_FOREVER);
+
+	ret = gpio_pca_series_reg_read_cache(dev, gpio_pca_series_2b_output_drive_strength,
+						 (uint8_t *)&reg_value);
+	if (ret) {
+		goto out;
+	}
+	reg_value = (reg_value & (~reg_cfg_mask)) | (strength << reg_cfg_shift);
+	ret = gpio_pca_series_reg_write(dev, gpio_pca_series_2b_output_drive_strength,
+					(uint8_t *)&reg_value);
+
+out:
+	k_sem_give(&data->lock);
+	return ret;
+}
+/* TODO support & drive strength */
+/**
+ * @}
+ * // gpio_pca_custom_api
+ */
+
+/*!
+ * @addtogroup gpio_pca_zephyr_gpio_api
+ * @{
+ */
+
+/**
+ * @brief configure gpio port.
+ *
+ * @note This API applies to all supported part no. Capabilities (
+ *       PCA_HAS_PULL and PCA_HAS_OUT_CONFIG) are evaluated and handled.
+ *
+ * @param dev
+ * @param pin
+ * @param flags
+ * @return int 0            if success
+ *             -ENOTSUP     if unsupported config flags are provided
+ */
+static int gpio_pca_series_pin_config(const struct device *dev,
+		gpio_pin_t pin, gpio_flags_t flags)
+{
+	const struct gpio_pca_series_config *cfg = dev->config;
+	struct gpio_pca_series_data *data = dev->data;
+	uint32_t reg_value;
+	int ret;
+
+	if (((flags & GPIO_INPUT) != 0) && ((flags & GPIO_OUTPUT) != 0)) {
+		return -ENOTSUP;
+	}
+
+	if (((flags & GPIO_SINGLE_ENDED) != 0U) &&
+		(cfg->part_cfg->flags & PCA_HAS_OUT_CONFIG) == 0U) {
+		return -ENOTSUP;
+	}
+
+	if (((flags & GPIO_PULL_UP) != 0) || ((flags & GPIO_PULL_DOWN) != 0)) {
+		if ((cfg->part_cfg->flags & PCA_HAS_PULL) == 0U) {
+			return -ENOTSUP;
+		}
+	} /* Can't do I2C bus operations from an ISR */
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	LOG_DBG("dev %s configure pin %d flag 0x%x", dev->name, pin, flags);
+
+	k_sem_take(&data->lock, K_FOREVER);
+
+	/* TODO: Only write 1 byte */
```

Comment:
Can you clarify what this comment means?

Reply:
Sure.
The config function only configures 1 pin, thus only 1 byte is changed per "register type". By only read-modify-write the changed byte, I2C transaction time can be optimized.

Reply:
Ah, I understand. We are writing all bytes of the register, but we only *need* to write one? Maybe add that information to the comment here, for future developers. This way the details of the optimization possible here aren't lost in the future

Reply:
I was planning to get it done in this pr. The problem here is that the current reg access api can only accept full reg (aka multi-byte). There has to be another api to modify a single byte.

Reply:
@danieldegrasse I'll leave a more detailed comments here, maybe try to optimize it in another pr, together with the drive strength api change.

This pr is large enough, I think I should focus on basic function and testing for now.